packages/adobe-sign/src/services/agreement.service.ts

```diff
@@ -0,0 +1,110 @@
+import { injectable } from "tsyringe";
+import { AdobeSignApi } from "../apis/adobe-sign/adobe-sign-api";
+import { Agreement } from "../apis/adobe-sign/models/agreement";
+import {AgreementPostData} from "../apis/adobe-sign/models/agreement-post-data";
+
+@injectable()
+export class AgreementService {
+    constructor(private adobeSignApi: AdobeSignApi) {}
+
+    public async getAgreement(id: string): Promise<Agreement> {
+        return await this.adobeSignApi.getAgreement(id);
+    }
+
+    //@todo replace with User Repo
+    public getUserData(userId: string): {[key: string]: string} {
+        console.log('retrieving user data for: ', userId);
+        return {
+            id: userId,
+            email: 'ngleapai@gmail.com',
+            firstName: 'Noah',
+            lastName: 'Leapai',
+            customField: 'some random value'
+        }
+    }
+
+    //@todo replace with Template Repo
+    public getTemplate(templateId: string): {[key: string]: any} {
+        console.log('retrieving template: ', templateId);
+        return {
+            id: templateId,
+            title: 'Template Title - API Test 008, interceptor test',
+            options: {
+                silenceEmails: true
+            },
+            mappings: [
+                {
+                    source: 'ods',
+                    sourceField: 'firstName',
+                    targetField: 'First.Name',
+                    defaultValue: '',
+                },
+                {
+                    source: 'ods',
+                    sourceField: 'lastName',
+                    targetField: 'Last.Name',
+                    defaultValue: '',
+                },
+                {
+                    source: 'ods',
+                    sourceField: 'customField',
+                    targetField: 'Text af442ffe-83fe-429d-a9bd-d00a2197ea3e',
+                    defaultValue: '',
+                },
+                {
+                    source: 'defaultValue',
+                    sourceField: null,
+                    targetField: 'Supplier.Type',
+                    defaultValue: 'domestic',
+                },
+            ]
+        }
+    }
+
+    public async createSigningUrlAgreement(templateId: string, userId: string): Promise<object> {
+        const newAgreementId = await this.createAgreement(templateId, userId);
+        const signingUrl = await this.getAgreementSigningUrls(newAgreementId);
+        return {
+            agreement_id: newAgreementId,
+            signing_url: signingUrl
+        }
+    }
+
+    public async createAgreement(templateId: string, userId: string): Promise<string> {
+        //@todo Check if agreement with user and template already exists in IN_PROGRESS state
```

Comment:
I didn't create a `getAgreementsByAsuriteId` function, you'll have to do that if you want to replace this in this PR.